kernel/core/commandfunctions.hpp

```diff
@@ -3077,6 +3077,33 @@ static CommandReturnValues cmd_delete(string& sCmd)
 }
 
 
+/////////////////////////////////////////////////
+/// \brief This static function clears all user
+/// defined variables.
+///
+/// \return void
+///
+/////////////////////////////////////////////////
+static void clear_variables()
+{
+    NumeRe::StringParser& _stringParser = NumeReKernel::getInstance()->getStringParser();
+    _stringParser.clearStringVar();
+
+    static std::vector<std::string> defaultVars = {"t", "x", "y", "z", "ncols", "nlens", "nlines", "nrows", "ans"};
+    Parser& _parser = NumeReKernel::getInstance()->getParser();
+    varmap_type mVariables = _parser.GetVar();
+
+    // check if user-defined variable is legacy variable and call is from terminal

```

Comment:
Comment does not make any sense. It should be something like `remove variables, which are not default variables and no temporary variables`